presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/QueryBuilder.java

```diff
@@ -103,24 +107,49 @@ public PreparedStatement buildSql(
             String catalog,
             String schema,
             String table,
+            List<ConnectorTableHandle> joinPushdownTables,
             List<JdbcColumnHandle> columns,
             TupleDomain<ColumnHandle> tupleDomain,
             Optional<JdbcExpression> additionalPredicate)
             throws SQLException
     {
+        boolean joinPushdownEnabled = checkIfJoinQuery(joinPushdownTables);
```

Comment:
I am going to push up a commit to your branch that further refactors this class, making it more readable, please take a look